src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.cpp

```diff
@@ -170,14 +205,22 @@ void KernelEmitter::emit_impl(const std::vector<size_t>& in,
     // todo: emit_impl is a const method, so we can't just push_back unused regs to the gp_regs_pool.
     //  we need a more elegant approach to avoid a full copy here
     auto local_gpr_pool = gp_regs_pool;
+    // we won't need indexes in both static and dynamic cases, since offsets are already calculated
     local_gpr_pool.push_back(static_cast<size_t>(reg_indexes.getIdx()));
-    local_gpr_pool.push_back(static_cast<size_t>(reg_const_params.getIdx()));
+    if (jcp.is_static) {
+        local_gpr_pool.push_back(static_cast<size_t>(reg_const_params.getIdx()));
+    }
     for (const auto& c : body) {
         const auto& emitter = c.first;
         std::vector<size_t> in_regs, out_regs;
         std::tie(in_regs, out_regs) = c.second;
-        if (auto tile_scheduler = std::dynamic_pointer_cast<TileSchedulerEmitter>(emitter))
+        if (auto tile_scheduler = std::dynamic_pointer_cast<TileSchedulerEmitter>(emitter)) {
+            // dynamic TileScheduler needs const runtime params
+            if (!jcp.is_static) {
+                in_regs.push_back(static_cast<size_t>(abi_param2.getIdx()));
+            }
```

Comment:
To align with comment
```suggestion
                in_regs.push_back(static_cast<size_t>(reg_const_params.getIdx()));
            }
```